jwst/stpipe/utilities.py

```diff
@@ -117,3 +136,99 @@ def islist_tuple(obj):
     Return True if `obj` is either a list or a tuple. False otherwise.
     """
     return isinstance(obj, tuple) or isinstance(obj, list)
+
+
+def all_steps():
```

Comment:
Is it safe to assume that the steps will be used in at least one pipeline?  If so, this might be a reasonable substitute:

```python
def all_steps():
    from jwst import pipeline

    steps = {}

    for pipeline_name in pipeline.__all__:
        pipeline_class = getattr(pipeline, pipeline_name)
        steps[pipeline_name] = pipeline_class
        for step_class in pipeline_class.step_defs.values():
            steps[step_class.__name__] = step_class

    return steps
```